### ISSUE TITLE
long hang on Firefox import

### ISSUE THREAD

**Issue by asadotzler** [brave/browser-laptop#4879]
**Did you search for similar issues before submitting this one?** Yes

**Describe the issue you encountered:** Importing Firefox data (history, bookmarks, and cookies) hung the Brave app for a good 30 seconds. Windows reported Brave as "(Not Responding)" for the duration. 

**Expected behavior:** Brave shouldn't hang. Also a progress indicator of some sort would be helpful if it's gonna take that long. 
- Platform (Win7, 8, 10? macOS? Linux distro?): Win10 
- Brave Version: 0.12.4
- Steps to reproduce:
  1. visit settings and import all types data from Firefox
  2. notice brave become unresponsive
  3. check task manager and see Not Responding for about 30 seconds
- Screenshot if needed:
- Any related issues: why no import passwords?

**Comment by polarity** [comment]
I have the same issue on OS X 10.10.5 with Brave 0.12.5. Brave hangs on Firefox import, never finishes and i have to kill the instance. Also nothing happened after reopening brave.

**Comment by bbondy** [comment]
@darkdh could we perhaps do things more async so it doesn't seem like a hang and maybe add a font-awesome animation like:

```
<i class="fa fa-spinner fa-spin fa-3x fa-fw"></i>
<span class="sr-only">Loading...</span>
```

**Comment by jkup** [comment]
I have a support ticket where a user (Win10 x64) has waited 30+ minutes without import finishing. Is this likely the same problem or something different?

**Comment by darkdh** [comment]
@jkup , did user import from firefox? and what option did user choose? (history/bookmarks/cookies )

**Comment by jkup** [comment]
Yes and bookmarks

**Comment by bbondy** [comment]
I can repro too w/ firefox imports but not chrome imports.

**Comment by alexwykoff** [comment]
I've repro'd with Chrome and large exported bookmarks files as well.  There is no system status on import progress.

**Comment by jaredbeck** [comment]
I encountered this issue on my first day of trying Brave. (Became unresponsive [beachball] and had to force quit.) Switching browsers without importing bookmarks and cookies is too big of a hurdle, so this is a blocker for me using Brave at all. Though negative, I hope this feedback is helpful! I'll try Brave in a few more months when it's more mature. Keep up the good work.

**Comment by bbondy** [comment]
Reverted due to automated test problems, and some other problems described in the PR that was reverted.

**Comment by nik9t** [comment]
same issue - Brave hangs on import of Firefox data - only option being to force quit. 3 attempts and sadly have to walk away from Brave until there is a solution to this problem

**Comment by luixxiul** [comment]
@nik9t it'll be solved in 0.13.1

**Comment by nik9t** [comment]
Thanks so much for letting me know - looking forward to that release!
